### PR TITLE
Modified section on thermostats and added section on choice of timestep

### PR DESCRIPTION
**Pull request by EfremBraun**

### CODE REVIEW

File: paper/basic_training.tex [hunk 1]
```diff
@@ -616,28 +620,52 @@ \subsubsection{Popular Thermostats}
...
 
-        The velocity rescaling thermostat is one of the simplest thermostats to implement, however, this thermostat is also one of the most non-physical thermostats as well.
+        The simple velocity rescaling thermostat is one of the easiest thermostats to implement, however, this thermostat is also one of the most non-physical thermostats as well.
 
          This thermostat relies on rescaling the momenta of the particles every $N$ timesteps based on their ratio of the instantaneous temperature to the target temperature\cite{thermostatAlgorithms2005}. 
          This leads to multiple issues when trying to sample data using this thermostat.
          First, this method is not reversible, there lacks a way for the particles to have knowledge of their previous thermal history.
          This makes any dynamical value impossible to measure (diffusion for example). 
          There is also no distribution of momenta; this thermostat does not sample the canonical distribution and cannot be used for sampling at thermal equilibrium.
+         The isokinetic (constant kinetic energy) ensemble samples the
```
Comment (davidlmobley) [inline paper/basic_training.tex]: A "one sentence per line" layout works really well with git diff logs as it makes it easier to see changes/edits, if you're willing to adjust your layout (or your editor) to accommodate that. Specifically, it makes revisions a lot easier to check in subsequent PRs.

File: paper/basic_training.tex [hunk 2]
```diff
@@ -533,7 +534,10 @@ \subsection{Main steps of a molecular dynamics simulation}
 It is highly recommended that best practices documents specific to this issue and to the type of system of interest be consulted.
 If such documents do not exist, it may be that a freely-available tool for constructing such a system does in fact exist.
 Examples include tools for constructing specific crystal structures, proteins, and lipid membranes.
-\todo[inline, color={green!20}]{JIM: Include some tools to cite here}
+\todo[inline, color={green!20}]{JIM: Include some tools to cite here.
+EB: Many tools are listed at
+\url{https://lammps.sandia.gov/prepost.html#builders}. My favorites include
```
Comment (davidlmobley) [inline paper/basic_training.tex]: Feel free to just go ahead and add. I'm a fan of packmol as well. Haven't used the others.

File: paper/basic_training.tex (hunk 1 above)
Comment (EfremBraun) [reply paper/basic_training.tex]: Sure. Fixing it soon.

File: paper/basic_training.tex [hunk 3]
```diff
@@ -616,28 +620,52 @@ \subsubsection{Popular Thermostats}
...
 
     \item \textbf{Hoover (Gaussian)}
+        \todo[inline, color={green!20}]{EB: I recommend referring to
+        this thermostat as the Gaussian Thermostat. I've seen it
+        referred to as the Hoover-Evans thermostat as well (I
+        haven't seen it called the Hoover thermostat, but I'm sure
+        somebody has called it that), but simply calling it the
+        Gaussian thermostat avoids
+        wading into the argument about who came up with it
+        first.}
```
Comment (davidlmobley) [inline paper/basic_training.tex]: I don't think I've seen Hoover either. @JIMonroe ?

File: paper/basic_training.tex [hunk 4]
```diff
@@ -616,28 +620,52 @@ \subsubsection{Popular Thermostats}
...
         
-        This thermostat is also an example of a velocity rescaling thermostat; however, the goal of the Gaussian thermostat is to ensure the change in
-        temperature $\Delta T$ is always 0 ($\Delta T \equiv 0$).
+        The goal of the Gaussian thermostat is to ensure the change in
+        instantaneous temperature $\Delta T$ is always 0 ($\Delta T \equiv 0$), which
+        is accomplished by modifying the force calculation with the form
+        $F = F_{interaction} + F_{Lagrange multiplier}$, where
+        $F_{interaction}$ is the standard interactions calculated during
+        the course of the simulation and $F_{Lagrange multiplier}$ keeps
```
Comment (davidlmobley) [inline paper/basic_training.tex]: I think this space in the subscript wouldn't show up properly?

File: paper/basic_training.tex [hunk 5]
```diff
@@ -616,28 +620,52 @@ \subsubsection{Popular Thermostats}
...
+        the kinetic energy constant.
```
Comment (davidlmobley) [inline paper/basic_training.tex]: Somewhat long sentence, a bit hard to parse. Split?

File: paper/basic_training.tex [hunk 6]
```diff
@@ -616,28 +620,52 @@ \subsubsection{Popular Thermostats}
...
 
-        Important information to know about this thermostat is that it is reversible and canonical; however, there are no temperature
-        fluctuations.
-        This once again will lead to issues when trying to sample any properties from the canonical ensemble.
-        Also, due to the nature of the thermostat preventing a change in temperature, whatever temperature
-        the system is at when the thermostat is applied, it will stay there. 
+        This thermostat is reversible and it samples the isokinetic
+        ensemble. Position-dependent equilibrium properties will
+        therefore be in agreement with the canonical ensemble (as
+        explained above), but dynamical properties will not.  due
```
Comment (davidlmobley) [inline paper/basic_training.tex]: "due" is lowercase at the start of a sentence, oops.

File: paper/basic_training.tex [hunk 7]
```diff
@@ -647,19 +675,31 @@ \subsubsection{Popular Thermostats}
...
         However, it should be noted that while allowing for temperature fluctuations, the Berendsen thermostat is not canonical nor is it reversible.
+        Its usage can lead to simulation artifacts, so it is not
+        recommended\cite{Braun:2018:arXiv}.
 
 \end{enumerate}
 
-Overall, \emph{none} of these deterministic thermostats are suitable for most production-level simulations, except perhaps during an initial equilibration phase, because they do not sample from the canonical ensemble~\cite{Shirts:2013:J.Chem.TheoryComput.}.
-Such thermostats can also lead to unexpected problems in systems with weak or poor coupling between degrees of freedom, such as the ``flying ice cube''~\cite{Harvey:1998:JCompChem} problem in some systems, and problems in alchemical free energy calculations where portions of the system are decoupled from the rest of the system (e.g. very rapid movement of atoms in the decoupled portion of the system).
+Overall, \emph{none} of these deterministic thermostats are suitable for
+most production-level simulations. though the Gaussian thermostat has
```
Comment (davidlmobley) [inline paper/basic_training.tex]: period where you should have a comma?

File: paper/basic_training.tex [hunk 8]
```diff
@@ -731,8 +787,28 @@ \subsection{Integrators}
...
+in the simulation box and the H-H bond vibration is the
+highest-frequency motion in the system, one can determine that with a
+force field harmonic force constant of approximately 500 N/m, the
+oscillation period will be 8 fs; thus, a 0.5 fs timestep can be used.
+For all-atom simulations with constraints on the high-frequency bonds,
+timesteps can be commonly increased to 2 fs; coarse-grained simulations
+with particles of higher mass and smaller force constants can have much
+larger timesteps.  After choosing a timestep, a test simulation should
+be run in the microcanonical ensemble to ensure that the choice of
+timestep yields dynamics that conserve energy.
```
Comment (davidlmobley) [inline paper/basic_training.tex]: OK with also adding something like, "To some extent, this depends on the choice of integrator; some integrators [perhaps reference Chodera/Fass recent work] may allow larger timesteps."

And do you think we should mention hydrogen mass repartitioning?

Comment (davidlmobley) [reply paper/basic_training.tex]: (I'm very glad you added this bit; it was a clear oversight on our part not to have it.)

**Comment by davidlmobley** [comment]
And please do add yourself as an author!

File: paper/basic_training.tex (hunk 4 above)
Comment (EfremBraun) [reply paper/basic_training.tex]: Right, fixed.

File: paper/basic_training.tex (hunk 5 above)
Comment (EfremBraun) [reply paper/basic_training.tex]: Fixed.

File: paper/basic_training.tex (hunk 6 above)
Comment (EfremBraun) [reply paper/basic_training.tex]: Fixed.

File: paper/basic_training.tex (hunk 7 above)
Comment (EfremBraun) [reply paper/basic_training.tex]: Fixed.

File: paper/basic_training.tex (hunk 2 above)
Comment (EfremBraun) [reply paper/basic_training.tex]: Fixed.

File: paper/basic_training.tex (hunk 8 above)
Comment (EfremBraun) [reply paper/basic_training.tex]: Definitely a good idea to add in a note that there are many ways to increase the timestep that can be used beyond that dictated by the smallest time period present in the system. Multiple-time-steps, hydrogen mass repartitioning, etc. can all be referenced as these kinds of methods.  I've added these.

**Comment by EfremBraun** [comment]
All requested changes made, and I added myself as an author (you can change me to whatever position in the list you want). Ready to pull.

**Comment by JIMonroe** [comment]
Looks great!  Thanks, @EfremBraun !  I may re-arrange some things in my next revision, but I like what you've added.

**Comment by davidlmobley** [comment]
Thanks! And on author order we'd decided on alphabetical, which probably means we put you first if you aren't there already. :)
